### DISCUSSION TITLE
Trouble getting Debugging Setup

### DISCUSSION THREAD

**Discussion by thomprl** [codefori/vscode-ibmi#2006] [General]
I'm having trouble getting debugging setup from Code For i.  I go through the walkthrough.  I have the debug extension installed, I made sure the PTF's are installed.  I go to the Setup Certificates and click Generate Certificates.  I get a pop-up

![image](https://github.com/codefori/vscode-ibmi/assets/17936568/cba18e03-1acc-4840-8f8b-2c573dedef63)

I click continue.

A few seconds later I get this message.

![image](https://github.com/codefori/vscode-ibmi/assets/17936568/ed176666-b66b-4b97-860f-f08f066566ae)

I look in this folder /QIBM/ProdData/IBMiDebugService/bin/cert and it does not exist.

**Comment by sebjulliand** [comment]
Hi @thomprl ,
the debug service configuration management is going to be improved in the next release. You can check out what's coming here: https://github.com/codefori/vscode-ibmi/pull/1990

With the element you gave here, I can only assume the certificates were generated by someone else than you, who used a different host name to connect to this system (i.e. the letters case was different). Since the password is based on this information, what was generated would not work for you.
The Debug Service Certificate is also supposed to be generated under `/QIBM/UserData/IBMiDebugService/certs`.

Long story short: if you can wait for the next release to re-configure the debug service, that would be great. Otherwise, you can at least delete any existing certificate under `/QIBM/UserData/IBMiDebugService/certs`, then switch to `Code for IBM i` pre-release version (as it contains some enhancement for the debug service already). From here, you can access the IBM i debugger view that can guide you through the certificate generation:
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/b7d724fe-d272-4eb1-87b5-099c3a84bc49)

**Comment by thomprl** [comment]
Ok, I was just trying it out.  Might want to update the walkthrough as it has a different folder for the certs than the one you listed.  The walkthrough has /QIBM/ProdData/IBMiDebugService/bin/certs.

![image](https://github.com/codefori/vscode-ibmi/assets/17936568/86b11399-05a9-4506-9de9-9d6e40d293be)

**Reply by sebjulliand** [reply]
This walkthrough is going away very soon, in favor of the IBM i debugger view 😅

**Reply by petterjohansson** [reply]
@sebjulliand . I have used the debug walktrough to start debug server, but that was as you mentioned removed in favor of the IBM i debugger view. When I try to start the debug server in this view:
![image](https://github.com/codefori/vscode-ibmi/assets/12429224/8dde54bb-f5d3-40c1-85e4-4ec2490bd40b)
![image](https://github.com/codefori/vscode-ibmi/assets/12429224/fc94d663-01fa-4dae-97c9-412f59933cac)
it fails. 

I currently start it manually through a ssh terminal when I need to debug:
<img width="718" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/12429224/c1af7356-36e5-40c9-a53b-adde90157dfb">
 
Any suggestions?

**Reply by sebjulliand** [reply]
At least the configuration process did its job since you can start the service using the shell script as it is (even though I'd rather start it using `SBMJOB` as described here: https://codefori.github.io/docs/developing/debug/#starting-the-debug-service-outside-of-code-for-ibm-i).

Can you share you Code for IBM i output after you try to start the service from VSCode? Let's see if there is something relevant in there.
